src/java/org/apache/cassandra/io/util/PathUtils.java

```diff
@@ -99,9 +99,31 @@ public final class PathUtils
     private static final Logger logger = LoggerFactory.getLogger(PathUtils.class);
     private static final NoSpamLogger nospam1m = NoSpamLogger.getLogger(logger, 1, TimeUnit.MINUTES);
 
-    private static Consumer<Path> onDeletion = path -> {};
     private static final boolean USE_NIX_RECURSIVE_DELETE = CassandraRelevantProperties.USE_NIX_RECURSIVE_DELETE.getBoolean();
 
+    private static volatile boolean DAEMON_SETUP_COMPLETED = false;
+
+    private static final CopyOption[] ATOMIC_MOVE_OPTIONS = {
+        StandardCopyOption.REPLACE_EXISTING,
+        StandardCopyOption.ATOMIC_MOVE
+    };
+
+    private static final CopyOption[] REPLACE_EXISTING_OPTIONS = {
+        StandardCopyOption.REPLACE_EXISTING
+    };
+
+    private static Consumer<Path> onDeletion = path -> {
+        if (DAEMON_SETUP_COMPLETED)
+            setDeletionListener(ignore -> {});
+        else if (logger.isTraceEnabled())
+            logger.trace("Deleting file during startup: {}", path);
+    };
+
+    public static void daemonSetupCompleted()
+    {
+        DAEMON_SETUP_COMPLETED = true;
+    }
+
```

Comment:
I think this may be extraneous; in CC4 `daemonSetupCompleted` is called by `CassandraDaemon.completeSetup`, but that's not the case in CC5. I think we just need the parts related to `CopyOption` here.